moto/shield/models.py

```diff
@@ -0,0 +1,202 @@
+            if resource_arns:
+                if len(resource_arns) > 1:
+                    raise ValidationException(
+                        "Error validating the following inputs: inclusionFilters.resourceArns"
+                    )
+                resource_protections = [
+                    protection
+                    for protection in self.protections.values()
+                    if protection.resource_arn == resource_arns[0]
+                ]
+            protection_names = inclusion_filters.get("ProtectionNames")
+            if protection_names:
+                if len(protection_names) > 1:
+                    raise ValidationException(
+                        "Error validating the following inputs: inclusionFilters.protectionNames"
+                    )
+                name_protections = [
+                    protection
+                    for protection in self.protections.values()
+                    if protection.name == protection_names[0]
+                ]
+            resource_types = inclusion_filters.get("ResourceTypes")
+            if resource_types:
+                if len(resource_types) > 1:
+                    raise ValidationException(
+                        "Error validating the following inputs: inclusionFilters.resourceTypes"
+                    )
+                type_protections = [
+                    protection
+                    for protection in self.protections.values()
+                    if protection.resource_type == resource_types[0]
+                ]
+            try:
+                protections = list(
+                    set.intersection(
+                        *(
+                            set(x)
+                            for x in [
+                                resource_protections,
+                                name_protections,
+                                type_protections,
+                            ]
+                            if x
+                        )
+                    )
+                )
+            except TypeError:
+                protections = []
+        else:
+            protections = list(self.protections.values())
+        return protections
+
+    def delete_protection(self, protection_id: str) -> None:
+        if protection_id in self.protections:
+            del self.protections[protection_id]
+            return
+        raise ResourceNotFoundException("The referenced protection does not exist.")
+
+    def list_tags_for_resource(self, resource_arn: str) -> List[Dict[str, str]]:
+        return self.tagger.list_tags_for_resource(resource_arn)["Tags"]
+
+    def tag_resource(self, resource_arn: str, tags: List[Dict[str, str]]) -> None:
+        self.tagger.tag_resource(resource_arn, tags)
+
+    def untag_resource(self, resource_arn: str, tag_keys: List[str]) -> None:
+        self.tagger.untag_resource_using_names(resource_arn, tag_keys)
+
+
+shield_backends = BackendDict(ShieldBackend, "ec2")
```

Comment:
I can't test this now, otherwise I would try it out. But can we change `"ec2"` to `"shield"`? Or will the tests fail if we do that?

Reply:
@bblommers It gives me **KeyError: 'aws'**, when I change it to "shield"